tests/suites/test_suite_asn1parse.function

```diff
@@ -735,15 +735,15 @@ void find_named_data( data_t *oid0, data_t *oid1, data_t *oid2, data_t *oid3,
 }
 /* END_CASE */
 
-/* BEGIN_CASE */
+/* BEGIN_CASE depends_on:!MBEDTLS_DEPRECATED_REMOVED */
 void free_named_data_null( )
 {
     mbedtls_asn1_free_named_data( NULL );
     goto exit; /* Silence unused label warning */
 }
 /* END_CASE */
 
-/* BEGIN_CASE */
+/* BEGIN_CASE depends_on:!MBEDTLS_DEPRECATED_REMOVED */
```

Comment:
```suggestion
/* BEGIN_CASE depends_on:!MBEDTLS_DEPRECATED_REMOVED:!MBEDTLS_DEPRECATED_WARNING */
```
To make the test_full_no_deprecated_deprecated_warning tests pass. 
Feel free to cherry-pick https://github.com/AndrzejKurek/mbedtls/commit/e1b37c198ccb30f2e19aad33c75ec8f45a816112.

Reply:
This should actually be `depends_on:MBEDTLS_TEST_DEPRECATED` as about to be officially documented via https://github.com/Mbed-TLS/mbedtls-docs/pull/62. But since it's equivalent, that's not a blocker.